src/test/java/booktopiahub/controller/OrderControllerTest.java

```diff
@@ -0,0 +1,235 @@
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/add-default-orders.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/delete-from-orders.sql")
+            );
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    void getAll_GivenOrdersInCatalog() throws Exception {
+        OrderDto orderFirst = new OrderDto()
+                .setId(1L)
+                .setUserId(1L)
+                .setTotal(BigDecimal.valueOf(50))
+                .setStatus(Order.Status.PENDING)
+                .setOrderDate(LocalDate.now())
+                .setOrderItems(Set.of())
+                .setShippingAddress("Kyiv, Ukraine");
+
+        OrderDto orderSecond = new OrderDto()
+                .setId(2L)
+                .setUserId(2L)
+                .setTotal(BigDecimal.valueOf(60))
+                .setStatus(Order.Status.PENDING)
+                .setOrderDate(LocalDate.now())
+                .setOrderItems(Set.of())
+                .setShippingAddress("Lviv, Ukraine");
+
+        List<OrderDto> expected = new ArrayList<>();
+        expected.add(orderFirst);
+        expected.add(orderSecond);
+
+        MvcResult result = mockMvc.perform(get("/orders")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        OrderDto[] actual = objectMapper.readValue(result.getResponse()
+                .getContentAsByteArray(), OrderDto[].class);
+        Assertions.assertEquals(2, actual.length);
+        Assertions.assertEquals(expected, Arrays.stream(actual).toList());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    void getOrderById() throws Exception {
+        Long orderId = 1L;
+        User user = new User();
+        user.setId(1L);
+        OrderDto expected = new OrderDto()
+                .setId(orderId)
+                .setUserId(user.getId())
+                .setTotal(BigDecimal.valueOf(50))
+                .setStatus(Order.Status.PENDING)
+                .setOrderDate(LocalDate.now())
+                .setOrderItems(Set.of())
+                .setShippingAddress("Kyiv, Ukraine");
```

Comment:
Please extract common expected models creations to separate method
lines 130-137 identical to 158-165

Recheck other classes